Realm/Tests/QueryTests.m

```diff
@@ -1307,7 +1307,7 @@ - (void)testDecimalQuery {
     [realm beginWriteTransaction];
     [DecimalObject createInRealm:realm withValue:@[@"-Inf"]];
     [DecimalObject createInRealm:realm withValue:@[@"Inf"]];
-    [DecimalObject createInRealm:realm withValue:@[@"123456789.123456789e12345"]];
```

Comment:
This number was overflowing and becoming negative with the old parser, while with the new parser it's correctly rounded to Inf. The test doesn't actually want an Inf value so I made the number fit within the representable range instead.